api/src/opentrons/protocol_api/instrument_context.py

```diff
@@ -3140,6 +3105,31 @@ def _handle_dispense_target(
         if isinstance(target, validation.PointTarget):
             return target.location, None, None
 
+    @property
+    def _last_tip_picked_up_from(self) -> Optional[labware.Well]:
+        """
+        .. deprecated:: 2.26
+           Use :py:obj:`ProtocolContext.get_last_tip_rack_well_used` instead.
+
+           If the pipette has a tip on it, returns the tip rack well it was picked up from.
+           Otherwise will return ``None``.
+        """
+        return self.get_last_tip_rack_well_used()
+
+    def get_last_tip_rack_well_used(self) -> Optional[labware.Well]:
+        """Returns the tip rack well the current tip has been picked up from.
+
+        If there is no tip currently on the pipette, this will return ``None``.
+        """
+        tip_rack_cores = self._core.get_last_well_tip_picked_up_from()
+        if tip_rack_cores is None:
+            return None
+        labware_core, well_core = tip_rack_cores
+        tip_rack_labware = self._core_map.get(labware_core)
+        return labware.Well(
+            parent=tip_rack_labware, core=well_core, api_version=self._api_version
+        )
```

Comment:
Forgot to address your question earlier-

> Does the way we reconstruct the Well object make sense?

I don't think it's a big issue. I believe we do construct new `Well` objects in a couple of other places. It just creates a slight inconvenience of not being able to use `is` identity checks on a different instance of the same well.

Did you try using an alternative like this? -
```suggestion
                return tip_rack_labware.wells_by_name()[well_core.get_name()]
```

Reply:
Incidentally, from memory leak investigations: It would be nice if we could always construct `Well` objects lazily instead of eagerly. They can easily get into the thousands, which I suspect it's a significant chunk of protocol memory usage. (I don't have reliable measurements handy to confirm this, though, so grain of salt.)

I vaguely remember the `is` thing being something that users problematically relied on, and we've tried to preserve it? But it just seems like such a losing battle, I'd support "breaking" it, if it helps.